src/main.cpp

```diff
@@ -2248,16 +2252,38 @@ DisconnectResult DisconnectBlock(const CBlock& block, const CBlockIndex* pindex,
         error("DisconnectBlock(): block and undo data inconsistent");
         return DISCONNECT_FAILED;
     }
+    CAddressAmountVec addressIndex;
+    CAddressUnspentVec addressUnspentIndex;
```

Comment:
I am ambivalent about the typedefs. I think that in this case I prefer just having the raw types, but I could go either way.

Reply:
@Eirik0, I took that halfway approach we discussed yesterday. I removed the `*Vec` types because they don't do much (save only the `vector<...>`) but I did keep the other typedefs (that are not in upstream) that begin with `pair<` because they do appear in many places, so seems like a useful abstraction.